packages/core/src/stateUtils.ts

```diff
@@ -84,7 +84,7 @@ type AdjList<TC, TE extends EventObject> = Map<
   Array<StateNode<TC, any, TE>>
 >;
 
-export const isLeafNode = (stateNode: StateNode<any, any, any>) =>
+export const isAtomicStateNode = (stateNode: StateNode<any, any, any>) =>
```

Comment:
Let's spike on this separately: https://github.com/davidkpiano/xstate/projects/1#card-34966341

Reply:
👍

Reply:
After thinking about this, I think it's best to keep it as `"atomic"` since SCXML only defines the opposite of a compound state as an "atomic" state and doesn't apply a special term to a non-final atomic state.

It makes sense to think of `<final>` state nodes as "subtypes" of atomic state nodes, but for practical reasons, we keep these as two separate types (`"atomic"` and `"final"`) and only make the relationship where it's needed (in `isAtomicStateNode`).